modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/CorrelatedNestedLoopJoinPlannerTest.java

```diff
@@ -160,4 +105,55 @@ public void testInvalidIndexExpressions() throws Exception {
 
         checkSplitAndSerialization(phys, publicSchema);
     }
+
+    /** */
+    @Test
+    public void testFilterPushDown() throws Exception {
+        IgniteSchema publicSchema = new IgniteSchema("PUBLIC");
+
+        publicSchema.addTable("T0", testTable(null));
+        publicSchema.addTable("T1", testTable(null));
+
+        String sql = "select * " +
+                "from t0 " +
+                "where t0.val > 'value' and exists (select * from t1 where t0.jid = t1.jid);";
+
+        IgniteRel phys = physicalPlan(sql, publicSchema);
+
+        System.out.println("+++ " + RelOptUtil.toString(phys));
+
+        assertNotNull(phys);
+
+        IgniteFilter filter = findFirstNode(phys, byClass(IgniteFilter.class));
+
+        assertNull("IgniteFilter must not be used.", filter);
+
+        IgniteTableScan tblScan = findFirstNode(
+                phys,
+                byClass(IgniteTableScan.class, (tbl) -> tbl.getTable().getQualifiedName().contains("T1")));
+
+        assertNotNull("Where condition must be push downed to IgniteTableScan.", tblScan.condition());
```

Comment:
It's an old, deprecated approach to check the plan. It's better to use `assertPlan` (it's less verbose, easier to understand and also has additional checks, like plan serialization)